server/protocols/jwt/src/main/java/org/apache/james/jwt/JwtConfiguration.java

```diff
@@ -21,13 +21,21 @@
 
 import java.util.Optional;
 
+import com.google.common.base.Preconditions;
+
 public class JwtConfiguration {
     private final Optional<String> jwtPublicKeyPem;
 
     public JwtConfiguration(Optional<String> jwtPublicKeyPem) {
+        Preconditions.checkState(validPublicKey(jwtPublicKeyPem), "The provided public key is not valid");
         this.jwtPublicKeyPem = jwtPublicKeyPem;
     }
 
+    private boolean validPublicKey(Optional<String> jwtPublicKeyPem) {
+        PublicKeyReader reader = new PublicKeyReader();
+        return !jwtPublicKeyPem.isPresent() || reader.fromPEM(jwtPublicKeyPem).isPresent();
```

Comment:
I find this boolean close hard to read....

Maybe a more functional style might be easier?

I propose:

```
PublicKeyReader reader = new PublicKeyReader();
jwtPublicKeyPem.map(value -> reader.fromPEM(Optional.of(value)).isPresent())
    .orElse(DEFAULT_VALID);
```